### DISCUSSION TITLE
API não deveria retornar BadRequest, Ok, Created?

### DISCUSSION THREAD

**Discussion by RanieryCG** [balta-io/7196#18] [Perguntas e Respostas] [answered]
Olá, tudo bem?

No curso a API está retornando um GenericCommandResult, por exemplo. Mas o certo não seria retornar um IActionResult com algum status? Exemplo: ok ou BadRequest

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia @RanieryCG seria sim, nos próximos cursos mostramos isto!

**Reply by RanieryCG** [reply]
Eu fiz assim, você teria alguma dica melhor, @andrebaltieri ?
![image](https://user-images.githubusercontent.com/6059830/138500629-de2bf74f-3bc7-4110-92fd-b82a695ecde9.png)

**Reply by andrebaltieri** [reply]
Está correto!!

**Comment by LucasBLs** [comment]
> Olá, tudo bem?
> 
> No curso a API está retornando um GenericCommandResult, por exemplo. Mas o certo não seria retornar um IActionResult com algum status? Exemplo: ok ou BadRequest

Aqui eu fiz dessa forma colega:
![image](https://user-images.githubusercontent.com/57819294/159129544-e2c475d2-208e-488e-9aa2-c0160d422b44.png)

**Comment by andrebaltieri** [comment]
No caso, retornamos o `StatusCode` que faz a conversão correta. O `IActionResult` poderia ser utilizado também, sem problemas :)
